tensornetwork/tn_keras/dense.py

```diff
@@ -41,10 +45,10 @@ class DenseDecomp(Layer):
     bias_initializer: Initializer for the bias vector.
 
   Input shape:
-    2D tensor with shape: `(batch_size, input_shape[-1])`.
+    nD tensor with shape: `(batch_size, input.shape[1:])`.
```

Comment:
Should be `input_shape` instead of `input.shape`, to match the usual Keras argument.
Also, tiny nit, but I think `N-D tensor` is more clear than `nD tensor`.

Reply:
Right. I changed it from input_shape to input.shape because in the included example, input_shape is a parameter passed to Keras specifying the size of the last dimension and thought this was confusing as an example. But I can change it back if you feel it is better.

Reply:
`input_shape` specifies the full input shape, not just the last dimension i.e. `input_shape[-1]` is the last dimension. You can also re-write to use an ellipsis. Check out this input specification example from Keras:

 Input shape:
    N-D tensor with shape: `(batch_size, ..., input_dim)`.
    The most common situation would be
    a 2D input with shape `(batch_size, input_dim)`.